docs/end_to_end_tutorials/principled_dev_practices.md

```diff
@@ -0,0 +1,42 @@
+# Principled Development Practices
+
+In order to develop your application, it can help to implement some principled development practices.
+
+We've boiled down the key practices into 3 pillars.
+
+We will also try to do a compare and contrast with traditional software engineering and ML development, highlighting similarities and differences.
+
+Here are the main pillars of principled development of LLM and RAG applications:
+
+```{toctree}
+---
+maxdepth: 1
+---
+/end_to_end_tutorials/dev_practices/observability.md
+/end_to_end_tutorials/dev_practices/evaluation.md
+/end_to_end_tutorials/dev_practices/monitoring.md
+```
+
+## The Development Pathway
+
+Start with a discovery phase of understanding your data and doing some identification of issues and corner cases as you interact with the system. 
+
+Try to formalize processes and evaluation methodology and develop your system, setting up tools for observability, debugging and experiment tracking, and eventually production monitoring.
+
+## Development Hurdles
+
+Here are some potential problems you may encounter when developing your LLM application which may lead to unsatisfactory results.
```

Comment:
Did we mean to add more details below yet?

Reply:
Not atm. We will add more over time.